packages/data/src/components/use-select/index.js

```diff
@@ -88,14 +88,24 @@ const renderQueue = createQueue();
  *
  * @return {Function}  A custom react hook.
  */
-export default function useSelect( _mapSelect, deps ) {
-	const isWithoutMapping = typeof _mapSelect !== 'function';
+export default function useSelect( mapSelect, deps ) {
+	const hasMappingFunction = 'function' === typeof mapSelect;
```

Comment:
would you be fine with not addressing that here? I would be very happy to look into this, but I'm trying to keep this PR focused on changes that preserve as much as possible

Reply:
Yeah, there is nothing actionable about this issue that would affect the current PR. But it's related to typing the hook, which is actually two very different hooks:
```ts
function useSelect( name: StoreDefinition | string ) : StoreSelectors;
function useSelect<T>( map: ( select: RegistrySelect ) => T, deps?: unknown[] ): T;
```
merged (unfortunately) into one.

Reply:
Regarding typing, I think we can leverage [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) on this, so I don't think it's much of a big deal. (Maybe I'm misinterpreting your comment though)

Reply:
What I mean that the types make it even more clear that the two overloads of `useSelect` are two very different hooks. We could also merge `useRegistry` and `useDispatch` together, because their arguments are different enough to reliably distinguish between the two usages, but we don't do that. I wish we could replace the `useSelect( name )` usages with
```js
useRegistry().select( name )
```
which is exactly the same thing, and return `useSelect` to supporting only the original use case.